## Question 1

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [42]:
#df.prglngth.value_counts().sort_index()
value_counts = df.prglngth.value_counts()
print "0-13 weeks", len(df[(df.prglngth >= 0) & (df.prglngth <= 13)])
print "14-26 weeks", len(df[(df.prglngth >= 14) & (df.prglngth <= 26)])
print "Over 26 weeks", len(df[(df.prglngth > 26)])
print "Total Number", len(df[df.prglngth])

0-13 weeks 3522
14-26 weeks 793
Over 26 weeks 9278
Total Number 13593


## Question 1

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [69]:
df.prglngth.value_counts().sort_index()

0       15
1        9
2       78
3      151
4      412
5      181
6      543
7      175
8      409
9      594
10     137
11     202
12     170
13     446
14      29
15      39
16      44
17     253
18      17
19      34
20      18
21      37
22     147
23      12
24      31
25      15
26     117
27       8
28      38
29      23
30     198
31      29
32     122
33      50
34      60
35     357
36     329
37     457
38     609
39    4744
40    1120
41     591
42     328
43     148
44      46
45      10
46       1
47       1
48       7
50       2
Name: prglngth, dtype: int64

## Question 1

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [83]:
counts = df.prglngth.value_counts().sort_index()

# group records by pregnancy length
def prglngth_group(prg_len):
    prg_len_labels = ["0-13 WEEKS", "14-26 WEEKS", "27+ WEEKS"]

    if (prg_len < 14):
        return prg_len_labels[0]
    if (prg_len >= 14 and prg_len < 27):
        return prg_len_labels[1]
    return prg_len_labels[2]
    
# total up the number in each group
prglngth_df = counts.groupby(prglngth_group).sum()

# add additional column
prglngth_df["TOTAL"] = counts.sum()

print prglngth_df

0-13 WEEKS      3522
14-26 WEEKS      793
27+ WEEKS       9278
TOTAL          13593
Name: prglngth, dtype: int64


## Question 1

###### Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [3]:
print "0-13 wks: " + str(sum(df.prglngth <= 13))
print "14-26 wks: " + str(sum((df.prglngth >= 14) & (df.prglngth <= 26)))
print ">27 wks: " + str(sum(df.prglngth >= 27))

0-13 wks: 3522
14-26 wks: 793
>27 wks: 9278


## Question 1

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [20]:
prglength_series = df.prglngth.value_counts().sort_index()
print " VALUE " + "COUNT"
print "--------------"
print " 0-13: " + str(sum(prglength_series[0:14]))
print "14-26: " + str(sum(prglength_series[14:27]))
print "27-50: " + str(sum(prglength_series[27:]))

 VALUE COUNT
--------------
 0-13: 3522
14-26: 793
27-50: 9278


## Question 1

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [100]:
print ("13 weeks or less: " + str(len(df[df.prglngth <= 13].index)))
print ("14 - 26 weeks: " + str(len(df[(df.prglngth >= 14) & (df.prglngth <=26)].index)))
print ("27 weeks or longer: " + str(len(df[df.prglngth >= 27].index)))

13 weeks or less: 3522
14 - 26 weeks: 793
27 weeks or longer: 9278


## Question 1

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [3]:
import numpy
print numpy.sum(df.prglngth <= 13)
print numpy.sum((df.prglngth > 13) & (df.prglngth < 27))
print numpy.sum(df.prglngth >= 27)

3522
793
9278


## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [57]:
#df.agepreg.value_counts().sort_index()
value_count = df.agepreg.value_counts()
print "15 and Under", len(df[(df.agepreg > 0) & (df.agepreg <= 15)]) #keep in mind ages are decimals, count 10.3 as 10
print "16-19 years", len(df[(df.agepreg > 15) & (df.agepreg < 20)])
print "20-24 years", len(df[(df.agepreg >= 20) & (df.agepreg < 25)])
print "25-29 years", len(df[(df.agepreg >= 25) & (df.agepreg < 30)])
print "30-44 years", len(df[(df.agepreg >= 30) & (df.agepreg < 45)])
print "Total Number", len(df.agepreg)

 15 and Under 173
16-19 years 3009
20-24 years 4246
25-29 years 3178
30-44 years 2635
Total Number 13593


## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [137]:
print 'value  ', '   total'
print '.         ', str(len(df.agepreg) - len(df.agepreg[(df.agepreg >= 0) & (df.agepreg < 45)]))
print '0-1999    ', str(len(df.agepreg[(df.agepreg >= 0) & (df.agepreg < 20)]))
print '2000-2499 ', str(len(df.agepreg[(df.agepreg >= 20) & (df.agepreg < 25)]))
print '2500-2999 ', str(len(df.agepreg[(df.agepreg >= 25) & (df.agepreg < 30)]))
print '3000-4499 ', str(len(df.agepreg[(df.agepreg >= 30) & (df.agepreg < 45)]))
print 'total:    ', len(df.agepreg)

value      total
.          352
0-1999     3182
2000-2499  4246
2500-2999  3178
3000-4499  2635
total:     13593


## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [70]:
df.agepreg.value_counts().sort_index()

10.33     1
10.50     1
10.83     1
10.91     1
11.08     1
11.41     1
11.50     1
11.75     2
12.08     1
12.25     1
12.41     2
12.50     1
12.66     2
12.75     1
12.83     2
12.91     1
13.00     2
13.08     1
13.16     1
13.25     4
13.33     1
13.50     3
13.58     2
13.66     5
13.75     4
13.83     5
13.91     5
14.00     5
14.08     7
14.16    12
         ..
40.50     6
40.58     4
40.66     4
40.75     2
40.83     4
40.91     4
41.00     2
41.08     2
41.16     2
41.25     6
41.33     1
41.41     2
41.58     4
41.75     3
41.83     1
41.91     4
42.08     2
42.16     1
42.25     1
42.33     1
42.41     1
42.50     2
42.58     2
42.75     2
43.00     3
43.16     1
43.25     4
43.58     3
43.91     1
44.08     1
Name: agepreg, dtype: int64

## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [94]:
counts = df.agepreg.value_counts().sort_index()

# The codebook divides age at pregnancy outcome into the following groups:
agepreg_labels = [
    "0-19 YEARS",
    "20-24 YEARS",
    "25-29 YEARS",
    "30-44 YEARS"
]

# group by each age group
def agepreg_group(age):
    if (age < 20):
        return agepreg_labels[0]
    if (age >= 20 and age < 25):
        return agepreg_labels[1]
    if (age >= 25 and age < 30):
        return agepreg_labels[2]
    return agepreg_labels[3]

# total up the number in each age group
agepreg_df = counts.groupby(agepreg_group).sum()

# add additional columns
agepreg_df["INAPPLICABLE"] = sum(df.agepreg.isnull())
agepreg_df["TOTAL"] = len(df.agepreg)

print agepreg_df

0-19 YEARS       3182
20-24 YEARS      4246
25-29 YEARS      3178
30-44 YEARS      2635
INAPPLICABLE      352
TOTAL           13593
Name: agepreg, dtype: int64


## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [4]:
print "Inapplicable: " + str(sum(df.agepreg.isnull()))
print "<20 years: " + str(sum(df.agepreg < 20))
print "20-24 years: " + str(sum((df.agepreg >= 20) & (df.agepreg < 25)))
print "25-29 years: " + str(sum((df.agepreg >= 25) & (df.agepreg < 30)))
print "30-44 years: " + str(sum((df.agepreg >= 30) & (df.agepreg < 45)))

Inapplicable: 352
<20 years: 3182
20-24 years: 4246
25-29 years: 3178
30-44 years: 2635


## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [101]:
print ("Under 20 years: " + str(len(df[df.agepreg <20].index)))
print ("20 - 24 years: " + str(len(df[(df.agepreg >= 20) & (df.agepreg <25)].index)))
print ("25 - 29 years: " + str(len(df[(df.agepreg >= 25) & (df.agepreg <30)].index)))
print ("30 - 44 years: " + str(len(df[(df.agepreg >= 30) & (df.agepreg <45)].index)))

Under 20 years: 3182
20 - 24 years: 4246
25 - 29 years: 3178
30 - 44 years: 2635


## Question 2

Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

In [90]:
df.agepreg.value_counts().sort_index()

10.33     1
10.50     1
10.83     1
10.91     1
11.08     1
11.41     1
11.50     1
11.75     2
12.08     1
12.25     1
12.41     2
12.50     1
12.66     2
12.75     1
12.83     2
12.91     1
13.00     2
13.08     1
13.16     1
13.25     4
13.33     1
13.50     3
13.58     2
13.66     5
13.75     4
13.83     5
13.91     5
14.00     5
14.08     7
14.16    12
         ..
40.50     6
40.58     4
40.66     4
40.75     2
40.83     4
40.91     4
41.00     2
41.08     2
41.16     2
41.25     6
41.33     1
41.41     2
41.58     4
41.75     3
41.83     1
41.91     4
42.08     2
42.16     1
42.25     1
42.33     1
42.41     1
42.50     2
42.58     2
42.75     2
43.00     3
43.16     1
43.25     4
43.58     3
43.91     1
44.08     1
Name: agepreg, dtype: int64

In order to compare this data to the code book, the sums of each age group needs to be calculated. The code below calculates the sums of the number of women pregnant between the ages of 20 - 24


In [91]:
ages = df.agepreg.value_counts().sort_index()

In [92]:
sum(ages[0:19.99])

3182

In [93]:
sum(ages[20:24.99])

4246

In [94]:
sum(ages[25:29.99])

3178

In [95]:
sum(ages[30:44.99])

2635

The data above matches the codebook's data for the specified age ranges (below)!! The ages that were described as inapplicable could not be summed to validate. 

AGEPREG ( 284-287 )

Variable Type : Preg recode

Description : Age at pregnancy outcome (recode)

value	label	 	Total
.	INAPPLICABLE	 	352
0-1999	UNDER 20 YEARS	 	3182
2000-2499	20-24 YEARS	 	4246
2500-2999	25-29 YEARS	 	3178
3000-4499	30-44 YEARS	 	2635
 	Total	 	13593


## Question 3

Create a new column named <tt>totalwgt_kg</tt> that contains birth weight in kilograms.  Compute its mean.  Remember that when you create a new column, you have to use dictionary syntax, not dot notation.

In [9]:
df['totalwgt_kg'] = df.totalwgt_lb * 0.453592
df.totalwgt_kg.mean()

3.2956309433503437

## Question 3

Create a new column named <tt>totalwgt_kg</tt> that contains birth weight in kilograms.  Compute its mean.  Remember that when you create a new column, you have to use dictionary syntax, not dot notation.

In [72]:
df['totalwgt_kg'] = df.birthwgt_lb / 2.2
df.totalwgt_kg.mean()

3.1055101877428464

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [24]:
df.rmarout6.value_counts().sort_index()
df.learnprg.value_counts().sort_index()
df.learnprg.mean()
df.learnprg.median()
df.learnprg.std()
df.pncarewk.value_counts().sort_index()
df.pncarewk.mean()
df.pncarewk.median()
df.pncarewk.std()

24.936835227466712

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [151]:
print 'pregordr mean ', df.pregordr.mean()
df.pregordr.value_counts().sort_index()

pregordr mean  2.34915029795


1     5033
2     3766
3     2334
4     1224
5      613
6      308
7      158
8       78
9       38
10      17
11       8
12       5
13       3
14       3
15       1
16       1
17       1
18       1
19       1
Name: pregordr, dtype: int64

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [73]:
df.birthord.mean()

1.8256449497157849

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [22]:
df.bfeedwks.value_counts().sort_index()
df.bfeedwks.mean()

465.20705176294075

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [7]:
df.agecon.value_counts().sort_index()
df.agecon.mean()

2420.70977709115

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [65]:
print 'BFEEDWKS DATA'
print 'mean: ',df.bfeedwks.mean()
print
# print df.bfeedwks.value_counts().sort_index()
print '<1    ',len(df.bfeedwks[df.bfeedwks == 0])
print '1-52  ',len(df.bfeedwks[(df.bfeedwks >= 1) & (df.bfeedwks <= 52)])
print '53+   ',len(df.bfeedwks[(df.bfeedwks >= 53) & (df.bfeedwks <= 993)])
print 'still ',len(df.bfeedwks[df.bfeedwks == 994])
print 'never ',len(df.bfeedwks[df.bfeedwks == 995])

BFEEDWKS DATA
mean:  465.207051763

<1     65
1-52   3805
53+    516
still  257
never  3355


## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

What percentage of pregnancies result in multiple births?

In [111]:
multipregs = df.multbrth.count()

percent = 100.0 * multipregs / len(df)
print "%.2f%%" % percent

1.20%


What's the average number of live births per pregnancy?

In [112]:
print df.nbrnaliv.mean(), "babies"

1.02241907262 babies


## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [74]:
df.hpagelb.value_counts().sort_index()


6       1
9       1
11      1
12      1
13      1
14      4
15     11
16     58
17    145
18    270
19    365
20    389
21    472
22    485
23    513
24    523
25    551
26    529
27    520
28    429
29    487
30    470
31    391
32    347
33    309
34    275
35    272
36    206
37    195
38    139
39    101
40    114
41     86
42     61
43     41
44     48
45     42
46     35
47     22
48     30
49     21
50     20
51      9
52      8
53      6
54      7
55      4
56      1
57      6
58      2
59      1
60      1
61      2
63      2
65      1
66      1
67      1
68      1
dtype: int64

In [75]:
df.hpagelb.mean()


27.848682754040293

In [76]:
df.hpagelb.median()

27.0

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [64]:
df[df.bfeedwks < 994].bfeedwks.mean() # Avg number of weeks breastfeeding, discounting those that never breastfed (code 995), and those currently breastfeeding (code 994)

28.965572275421795

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [8]:
print 'Average weights in pounds:'
print 'MALE\t', df[df.babysex == 1].totalwgt_lb.mean()
print 'FEMALE\t', df[df.babysex == 2].totalwgt_lb.mean()

Average weights in pounds:
MALE	7.41140948746
FEMALE	7.11552604985


## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [7]:
bfeedwks = df.bfeedwks
# filter out responses that did not specifically indicate a breast-feeding length. See documentation at tinyurl.com/zn9kw2d
bfeedwks_filtered = bfeedwks[(bfeedwks.notnull()) & (bfeedwks < 994)];
print "mean breastfeed weeks: " + str(bfeedwks_filtered.mean())
print "quartiles:"
print bfeedwks_filtered.quantile([.25, .5, .75])


mean breastfeed weeks: 28.9655722754
quartiles:
0.25     9
0.50    22
0.75    43
dtype: float64


## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [83]:
print "Use of maternity leave"
print "----------------------"
print "                                               inapplicable: " + str(len(df.maternlv) - len(df.maternlv[df.maternlv < 5]))
print "                              not employed during pregnancy: " + str(len(df.maternlv[df.maternlv == 0]))
print "                                        took leave from job: " + str(len(df.maternlv[df.maternlv == 1]))
print "did not take leave from job (job schedule or self-employed): " + str(len(df.maternlv[df.maternlv == 2]))
print "          did not take leave from job (not offered/allowed): " + str(len(df.maternlv[df.maternlv == 3]))
print "                did not take leave from job (other reasons): " + str(len(df.maternlv[df.maternlv == 4]))

Use of maternity leave
----------------------
                                               inapplicable: 10402
                              not employed during pregnancy: 1355
                                        took leave from job: 1244
did not take leave from job (job schedule or self-employed): 146
          did not take leave from job (not offered/allowed): 93
                did not take leave from job (other reasons): 353


## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [104]:
#This is the age of the respondent at the time of the interview
df.ager.value_counts().sort_index()

15      7
16     16
17     36
18     76
19    111
20    140
21    187
22    277
23    345
24    359
25    416
26    453
27    502
28    493
29    586
30    577
31    618
32    665
33    640
34    636
35    638
36    740
37    695
38    657
39    575
40    679
41    636
42    594
43    650
44    589
Name: ager, dtype: int64

In [105]:
#The mean age of respondents is 33
df.ager.mean()

33.470904141837714

In [106]:
#And the standard deviation in the age of respondents is 6.616
df.ager.std()

6.715705627398683

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [98]:
df.bfeedwks.value_counts().sort_index()

0        65
1       122
2       182
3        95
4       288
5        33
6       138
7        18
8        45
9       329
10       24
11        5
12       14
13      438
14        4
15        2
16        1
17      323
18        6
19        1
22      175
24        2
26      413
30      180
31        1
32        2
33        1
35      191
36        1
39      180
       ... 
61       68
65       45
69       22
74       11
78       95
82        7
87       15
91        6
95       10
100       1
104      87
108       4
112       1
113       4
117       1
121       1
126       1
130      19
134       1
139       2
143       1
152       4
156      29
160       2
182       3
199       1
208       8
303       1
994     257
995    3355
Name: bfeedwks, dtype: int64

In [99]:
df.bfeedwks.mean()

465.20705176294075

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [8]:
max(df.nbrnaliv)

5.0

## Question 4

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

In [7]:
df.agepreg.mean()

24.688151197038128

## Question 5

Count the number of live births with <tt>birthwgt_lb</tt> between 9 and 95 pounds (including both).  The result should be 798 

In [28]:
len(live[(live.birthwgt_lb >= 9) & (live.birthwgt_lb <= 95)]) #95 is an aweful lot...

798

## Question 5

Count the number of live births with <tt>birthwgt_lb</tt> between 9 and 95 pounds (including both).  The result should be 798 

In [144]:
len(live[(live.birthwgt_lb > 8) & (live.birthwgt_lb < 96)])

798

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [58]:
f_prglngth = firsts.prglngth.mean()
o_prglngth = others.prglngth.mean()
diff = (f_prglngth - o_prglngth) * 7 * 24 #weeks to days to hours
print diff, "hours"

13.1102608186 hours


## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [148]:
abs(firsts.prglngth.mean() - others.prglngth.mean())*7*24

13.11026081862832

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [30]:
lenf = firsts.prglngth.mean()
leno = others.prglngth.mean()
hrsinwk = 24*7
diff = (lenf - leno)*hrsinwk
lenf
leno
diff

13.11026081862832

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [73]:
fml = firsts.prglngth.mean()
oml = others.prglngth.mean()
(fml-oml)*7*24

13.110260818628319

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [84]:
firsts.prglngth.mean()

38.60095173351461

In [85]:
others.prglngth.mean()

38.52291446673706

In [86]:
difference = firsts.prglngth.mean()-others.prglngth.mean()
difference

0.07803726677754952

In [87]:
difference * 7 * 24

13.11026081862832

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [72]:
first_avg = firsts.prglngth.mean()
other_avg = others.prglngth.mean()

first_avg - other_avg

0.07803726677754952

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [114]:
firstlen = firsts.prglngth.mean()
firstlen

38.60095173351461

In [115]:
otherlen = others.prglngth.mean()
otherlen

38.52291446673706

In [116]:
#Compute the difference between first babies and other babies
#Then convert to hours
diff = (firstlen - otherlen) * 7 * 24
diff

13.11026081862832

It appears that first babies tend to be born an average of 13 hours after second babies, so if they are late, they are not late by much. 

## Question 6

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.

In [107]:
firsts.prglngth.mean()

38.60095173351461

In [108]:
others.prglngth.mean()

38.52291446673706

In [109]:
firsts.prglngth.mean() - others.prglngth.mean()

0.07803726677754952

## Clarifying Questions

Use this space to ask questions regarding the content covered in the reading. These questions should be restricted to helping you better understand the material. For questions that push beyond what is in the reading, use the next answer field. If you don't have a fully formed question, but are generally having a difficult time with a topic, you can indicate that here as well.

I think I generally understand this. Basically, there is an "ultimate dataset" called a data frame. This contains coded data which can be accessed through pre-defined label. Data types to deal with throughout this are the data frames, dictionary-like elements, collections, and series. 

No specific questions, but it would be nice to go over in class the best way to do each part (most efficient, readable, etc.)

I am confused about recodes, as comparing numbers computed from the whole data set to the data set itself seems very redundant. Do we take a small sample of the entire raw data to calculate the recodes? 

## Enrichment Questions

Use this space to ask any questions that go beyond (but are related to) the material presented in this reading. Perhaps there is a particular topic you'd like to see covered in more depth. Perhaps you'd like to know how to use a library in a way that wasn't show in the reading. One way to think about this is what additional topics would you want covered in the next class (or addressed in a followup e-mail to the class). I'm a little fuzzy on what stuff will likely go here, so we'll see how things evolve.

I'm wondering how typical it is to find datasets in the format here. I've done a fair amount of stuff with aid-data resources, and they are much less sophisticated than this...we're talking CSVs as the most hi-tech translation of the data. 

Just as a general note, it was cool to do something with population data, and it would be great to do more peoples analysis in the future. An element that was not explored here was doing cross comparisons (correlations for example), which I'm sure will come later. Significance would also be cool to learn more about.

I am curious if whether we will always be given data sets in such convenient form to use in this class, or whether we will build data sets ourselves.

I know it's only chapter one so I don't have too many questions, but the one I was wondering about as I was doing this assignment was how to figure out the value of the calculations.  Like, we could calculate so many values from this dataset, how do we know what is important?

What is the difference between section B of the Raw Variables and Section E in the codebook? I'm just curious about what context I'm missing for the different variables. 


This chapter mentioned that the NSFG intentionally oversampled Hispanics, African-Americans, and teenagers. How can this be addressed if you want to look at a statistically-correct distribution over the entire population?

How would you deal with situations where the data doesn't accurately represent the population, e.g., certain demographics are more represented than others? Are there tools for analyzing or cleaning the data?

This is something that I'm a little unclear on in general, but I'm wondering what sorts of data science scenarios it's benficial/the "right" answer is to use a graph, and what questions are better answered using summary statistics/numbers. Is it always helpful to visualize? Are there times when it's better/acceptable not to?

How do you validate data if no published results exist for comparison? 

## Additional Resources / Explorations

If you found any useful resources, or tried some useful exercises that you'd like to report please do so here. Let us know what you did, what you learned, and how others can replicate it.

I liked having pandas.pydata.org to read more on the different possible methods for a dataframe